src/Microsoft.AspNet.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -64,9 +64,9 @@ public BodyModelBinder()
                     // system to skip other model binders and never to fall back.
                     return new ModelBindingResult(modelBindingKey);
                 }
-
-                var valueProviderResult = new ValueProviderResult(rawValue: model);
-                bindingContext.ModelState.SetModelValue(modelBindingKey, valueProviderResult);
+                
+                // We don't have anything useful to put in model state.
```

Comment:
Huh? We changed this code not that long ago to explicitly get the `model` into the `ModelStateDictionary`. Thought one of the intents here was to separate `ValueProviderResult` from `ModelState`. Instead seem to be reproducing the new `ValueProviderResult` restrictions in `ModelState`.

Put another way, I expected `bindingContext.ModelState.SetModelValue(modelBindingKey, nonStringIshValue: model)`.